modules/userId.js

```diff
@@ -253,14 +260,36 @@ export function requestBidsHook(fn, reqBidsConfigObj) {
       }
     }
   }
+}
 
+/**
+ * Hook is executed before adapters, but after consentManagement. Consent data is requied because
+ * this module requires GDPR consent with Purpose #1 to save data locally.
+ * The two main actions handled by the hook are:
+ * 1. check gdpr consentData and handle submodule initialization.
+ * 2. append user id data (loaded from cookied/html or from the getId method) to bids to be accessed in adapters.
+ * @param {Object} reqBidsConfigObj required; This is the same param that's used in pbjs.requestBids.
+ * @param {function} fn required; The next function in the chain, used by hook.js
+ */
+export function requestBidsHook(fn, reqBidsConfigObj) {
+  // initialize submodules only when undefined
+  initializeSubmodulesAndExecuteCallbacks();
   // pass available user id data to bid adapters
   addIdDataToAdUnitBids(reqBidsConfigObj.adUnits || getGlobal().adUnits, initializedSubmodules);
-
   // calling fn allows prebid to continue processing
   return fn.call(this, reqBidsConfigObj);
 }
 
+/**
+ * This function will be exposed in global-name-space so that userIds stored by Prebid UserId module can be used by external codes as well.
+ * Simple use case will be passing these UserIds to A9 wrapper solution
+ */
+function getUserIds() {
```

Comment:
@pm-harshad-mane Should this function not have a callback function as argument which will be called when user id's are ready. Id systems make async calls to get user ids. This function may return empty object or undefined.

Reply:
Hello @jaiminpanchal27 ,
my expectation from the API is to return whatever data is available at the moment.
It will be a good feature to have a call-back on getting all ids, but for now, it is not a requirement. Also, as userId module is caching the data, maybe on the second iteration of execution the new API will return all ids so it is sufficient for now.